backend/Watchdog.Core/Watchdog.Core.BLL/Services/IssueService.cs

```diff
@@ -139,22 +153,61 @@ private async Task<ICollection<IssueMessage>> GetIssueMessagesAsync()
         private async Task<int> GetTotalHitsAsync<T>() where T : class
         {
             var totalHits = await _client.CountAsync<T>();
-            return (int) totalHits.Count;
+            return (int)totalHits.Count;
         }
 
-        public async Task UpdateAssignee(UpdateAssigneeDto assigneeDto)
+        public Task UpdateAssigneeAsync(UpdateAssigneeDto assigneeDto)
         {
-            var issueResponse = await _client.GetAsync<Issue>(assigneeDto.IssueId);
-            if (!issueResponse.IsValid)
+            if (assigneeDto is null)
+            {
+                throw new System.ArgumentNullException(nameof(assigneeDto));
+            }
+
+            return UpdateAssigneeInternalAsync(assigneeDto);
+
+        }
+        private async Task UpdateAssigneeInternalAsync(UpdateAssigneeDto assigneeDto)
+        {
+            if (!(await _client.GetAsync<Issue>(assigneeDto.IssueId)).IsValid)
             {
                 throw new KeyNotFoundException("Issue doesn't exist");
             }
 
-            var issue = issueResponse.Source;
+            var oldMembers = await _context.AssigneeMembers
+                .Where(a => a.IssueId == assigneeDto.IssueId)
+                .ToListAsync(); // assignee members in db
+            var membersToAdd = assigneeDto.Assignee.MemberIds
+                .Except(oldMembers.Select(a => a.MemberId)); // members in db - members in dto
+            var membersToDelete = oldMembers
+                .Where(m => !assigneeDto.Assignee.MemberIds.Any(id => m.MemberId == id)); // members in dto - members in db
+
+            await _context.AssigneeMembers.AddRangeAsync(
+                membersToAdd.Select(id => new AssigneeMember
+                {
+                    IssueId = assigneeDto.IssueId,
+                    MemberId = id
+                }));
+
+            _context.AssigneeMembers.RemoveRange(membersToDelete);
+
+            var oldTeams = await _context.AssigneeTeams
+                .Where(a => a.IssueId == assigneeDto.IssueId)
+                .ToListAsync(); // assignee teams in db
+            var teamsToAdd = assigneeDto.Assignee.TeamIds
+                .Except(oldTeams.Select(a => a.TeamId)); // teams in db - teams in dto
+            var teamsToDelete = oldTeams
+                .Where(t => !assigneeDto.Assignee.TeamIds.Any(id => t.TeamId == id)); // teams in db - teams in dto
+
+            await _context.AssigneeTeams.AddRangeAsync(
+                teamsToAdd.Select(id => new AssigneeTeam
+                {
+                    IssueId = assigneeDto.IssueId,
+                    TeamId = id
+                }));
+
+            _context.AssigneeTeams.RemoveRange(teamsToDelete);
 
-            issue.Assignee = assigneeDto.Assignee;
-            await _client.UpdateAsync<Issue, Issue>(assigneeDto.IssueId, descriptor => descriptor
-                        .Doc(issue));
+            await _context.SaveChangesAsync();
         }
```

Comment:
Is this method invokes every time the assignee adds to the issue? Or they assigns when the dialog window closes?